Issue when using @IndexPrefixes annotation

Hello,

I have a simple index like this : 

```java
import org.springframework.data.annotation.Id;

@Document(indexName = "geo")
public class GeoDocument {

    @Id
    private String id;

    @Field(type = FieldType.Text, indexPrefixes = @IndexPrefixes(minChars = 2, maxChars = 10))
    private String code;

}
```

**Current behavior**

The execution of my application fails with following stacktrace : 

```
Caused by: co.elastic.clients.json.JsonpMappingException: Error deserializing co.elastic.clients.elasticsearch._types.mapping.TextProperty: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TextIndexPrefixes.minChars' (JSON path: properties.code.index_prefixes) (in object at line no=1, column no=110, offset=109)
	at co.elastic.clients.json.JsonpMappingException.from0(JsonpMappingException.java:134) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpMappingException.from(JsonpMappingException.java:121) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:236) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:166) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:157) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:79) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:78) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:231) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:166) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:157) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.BuildFunctionDeserializer.deserialize(BuildFunctionDeserializer.java:47) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpDeserializerBase$StringMapDeserializer.deserialize(JsonpDeserializerBase.java:349) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpDeserializerBase$StringMapDeserializer.deserialize(JsonpDeserializerBase.java:333) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:78) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:214) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:166) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.util.WithJsonObjectBuilderBase.withJson(WithJsonObjectBuilderBase.java:54) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.WithJson.withJson(WithJson.java:57) ~[elasticsearch-java-9.4.2.jar:na]
	at org.springframework.data.elasticsearch.client.elc.TypeUtils.lambda$typeMapping$0(TypeUtils.java:480) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at co.elastic.clients.elasticsearch._types.mapping.TypeMapping.of(TypeMapping.java:138) ~[elasticsearch-java-9.4.2.jar:na]
	at org.springframework.data.elasticsearch.client.elc.TypeUtils.typeMapping(TypeUtils.java:480) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at org.springframework.data.elasticsearch.client.elc.RequestConverter.indicesCreateRequest(RequestConverter.java:254) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.doCreate(IndicesTemplate.java:147) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.createWithMapping(IndicesTemplate.java:135) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.createIndexAndMappingIfNeeded(SimpleElasticsearchRepository.java:94) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.<init>(SimpleElasticsearchRepository.java:87) ~[spring-data-elasticsearch-6.1.0.jar:6.1.0]
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:207) ~[spring-beans-7.0.6.jar:7.0.6]
	... 60 common frames omitted
Caused by: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'TextIndexPrefixes.minChars'
	at co.elastic.clients.util.ApiTypeHelper.checkNotNull(ApiTypeHelper.java:81) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:91) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.elasticsearch._types.mapping.TextIndexPrefixes.<init>(TextIndexPrefixes.java:71) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.elasticsearch._types.mapping.TextIndexPrefixes$Builder.build(TextIndexPrefixes.java:165) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.elasticsearch._types.mapping.TextIndexPrefixes$Builder.build(TextIndexPrefixes.java:123) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:80) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:78) ~[elasticsearch-java-9.4.2.jar:na]
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:214) ~[elasticsearch-java-9.4.2.jar:na]
	... 92 common frames omitted
```

**Workaround**

Using `@Mapping` annotation :

```java
@Mapping(mappingPath = "mapping.json")
```

With the following content : 

```json
{
  "properties": {
    "id": {
      "type": "text",
      "fields": {
        "keyword": {
          "type": "keyword",
          "ignore_above": 256
        }
      }
    },
    "code": {
      "type": "text",
      "index_prefixes": {
        "min_chars": 2,
        "max_chars": 10
      }
    }
  }
}
```

Minimal repo for bug reproduction : https://github.com/ii02735/spring-data-es_index_prefixes_bug

Thank you in advance for your help.

--- a/pom.xml
+++ b/pom.xml
@@ -21,8 +21,8 @@
 		<springdata.commons>4.2.0-SNAPSHOT</springdata.commons>
 
 		<!-- version of the ElasticsearchClient	-->
-		<elasticsearch-java>9.4.2</elasticsearch-java>
-		<elasticsearch-rest-client>9.4.2</elasticsearch-rest-client>
+		<elasticsearch-java>9.4.3</elasticsearch-java>
+		<elasticsearch-rest-client>9.4.3</elasticsearch-rest-client>
 
 		<hoverfly>0.20.2</hoverfly>
 		<log4j>2.25.4</log4j>

--- a/src/main/antora/modules/ROOT/pages/elasticsearch/elasticsearch-new.adoc
+++ b/src/main/antora/modules/ROOT/pages/elasticsearch/elasticsearch-new.adoc
@@ -1,6 +1,11 @@
 [[new-features]]
 = What's new
 
+[[new-features.6-2-0]]
+== New in Spring Data Elasticsearch 6.2
+
+* Upgrade to Elasticsearch 9.4.3
+
 [[new-features.6-1-0]]
 == New in Spring Data Elasticsearch 6.1
 

--- a/src/main/antora/modules/ROOT/pages/elasticsearch/versions.adoc
+++ b/src/main/antora/modules/ROOT/pages/elasticsearch/versions.adoc
@@ -6,6 +6,7 @@ The following table shows the Elasticsearch and Spring versions that are used by
 [cols="^,^,^,^",options="header"]
 |===
 | Spring Data Release Train | Spring Data Elasticsearch | Elasticsearch | Spring Framework
+| 2026.1 | 6.2.x | 9.4.3 | 7.0.x
 | 2026.0 | 6.1.x | 9.4.2 | 7.0.x
 | 2025.1 | 6.0.x | 9.2.2 | 7.0.x
 | 2025.0 | 5.5.xfootnote:oom[Out of maintenance] | 8.18.1 | 6.2.x

--- a/src/test/java/org/springframework/data/elasticsearch/core/index/MappingBuilderIntegrationTests.java
+++ b/src/test/java/org/springframework/data/elasticsearch/core/index/MappingBuilderIntegrationTests.java
@@ -280,6 +280,12 @@ public abstract class MappingBuilderIntegrationTests extends MappingContextBaseT
 		operations.indexOps(FieldAliasEntity.class).createWithMapping();
 	}
 
+	@Test // #3298
+	@DisplayName("should write index prefixes with default value")
+	void shouldWriteIndexPrefixesWithDefaultValue() {
+		operations.indexOps(TextIndexPrefixesEntity.class).createWithMapping();
+	}
+
 	// region Entities
 	@Document(indexName = "#{@indexNameProvider.indexName()}")
 	static class Book {
@@ -929,6 +935,15 @@ public abstract class MappingBuilderIntegrationTests extends MappingContextBaseT
 		@Field(type = Text) private String someText;
 		@Nullable
 		@Field(type = Text) private String otherText;
+	}
+
+	@Document(indexName = "#{@indexNameProvider.indexName()}")
+	private static class TextIndexPrefixesEntity {
+		@Id
+		@Nullable private String id;
+		@Nullable
+		// the min value is set to the default
+		@Field(type = Text, indexPrefixes = @IndexPrefixes(maxChars = 10)) private String someText;
 	}
 
 	// endregion

--- a/src/test/resources/testcontainers-elasticsearch.properties
+++ b/src/test/resources/testcontainers-elasticsearch.properties
@@ -15,7 +15,7 @@
 #
 #
 sde.testcontainers.image-name=docker.elastic.co/elasticsearch/elasticsearch
-sde.testcontainers.image-version=9.4.2
+sde.testcontainers.image-version=9.4.3
 #
 #
 # needed as we do a DELETE /* at the end of the tests, will be required from 8.0 on, produces a warning since 7.13

